### PR TITLE
[KEYCLOAK-6147] Include Nonce in OIDC authentication 

### PR DESCRIPTION
**Pull request by jayblanc**
Include Nonce in OIDC authentication request to avoid problem with FranceConnect

### CODE REVIEW

File: services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java [hunk 1]
```diff
@@ -319,6 +320,16 @@ protected UriBuilder createAuthorizationUrl(AuthenticationRequest request) {
         if (acr != null) {
             uriBuilder.queryParam(OAuth2Constants.ACR_VALUES, acr);
         }
+
+        String nonce = request.getAuthenticationSession().getClientNote(OIDCLoginProtocol.NONCE_PARAM);
+        if (nonce == null || nonce.isEmpty()) {
+            nonce = UUID.randomUUID().toString();
+            request.getAuthenticationSession().setClientNote(OIDCLoginProtocol.NONCE_PARAM, nonce);
+        }
+        if (nonce != null) {
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java]: It couldn't happen that "nonce" will be null if I read correctly? Isn't this "if" check unecessary? Besides that, PR looks good to me.

**Review by mposolda — CHANGES_REQUESTED** [review]
PR looks good. I just left one very minor comment here.

**Comment by jayblanc** [comment]
You're right, It is because I made a version that include checkox in the
Identity provider config and I downgrade to this simplest version for the
first PR letting this useless check on non null.
(In the case of the config, it could be null).
By the way, do you have any advice for testing the admin app config because
I also made the version that include config but I did'nt made the PR
because of missing testing) ?
Best regards, Jérôme.

Le lun. 19 mars 2018 à 08:07, Marek Posolda <notifications@github.com> a
écrit :

**Comment by mposolda** [comment]
PR looks completely messed now. Could you just rebase from latest master and add just single commit with your change on top?

**Comment by jayblanc** [comment]
I made a rebase but i don't know why it make all that mess !! I will make
it cleaner asap. sorry.

Le lun. 19 mars 2018 à 19:29, Marek Posolda <notifications@github.com> a
écrit :

**Comment by jayblanc** [comment]
I'm going to make a new pull request without mess.

**Comment by jayblanc** [comment]
The new pull request is #5082 (https://github.com/keycloak/keycloak/pull/5082)

**Comment by mposolda** [comment]
@jayblanc Thanks! I've merged new PR. I am closing this one then.

**Pull request closed without merging**